internal/http/delete.go

```diff
@@ -1,37 +1,15 @@
 package http
 
 import (
-	"io/ioutil"
 	"net/http"
 )
 
 // Delete performs a DELETE request
 func (client Client) Delete(url string, headers RequestHeaders) ([]byte, ResponseHeaders, error) {
-	headers = client.AppendDefaultHeaders(headers)
-
-	req, err := http.NewRequest(http.MethodDelete, url, nil)
-	if err != nil {
-		return nil, ResponseHeaders{}, err
-	}
-
-	for _, header := range headers {
-		req.Header.Set(header.Name, header.Value)
-	}
-
-	resp, err := client.Do(req)
-	if err != nil {
-		return nil, ResponseHeaders{}, err
-	}
-
-	err = HandleStatus(resp.StatusCode)
-	if err != nil {
-		return nil, ResponseHeaders{}, err
-	}
-
-	body, err := ioutil.ReadAll(resp.Body)
+	responseBody, responseHeaders, err := client.Do(http.MethodDelete, url, headers, nil)
 	if err != nil {
 		return nil, ResponseHeaders{}, err
 	}
 
-	return body, ResponseHeaders{resp.Header}, nil
+	return responseBody, responseHeaders, nil
```

Comment:
This may as well just return "client.Do(http.MethodDelete, url, headers, nil)" as the method it calls does that anyway, no need to do the error nil check.

We could also extract the error handling to a method: 

```
func noContentError(err error) (([]byte, ResponseHeaders, error)) {
    return nil, ResponseHeaders{}, err
}
```

Reply:
yeah I don't love immediately returning the output of other functions, its chance that their return methods line up, not design, and so in future if I want to change the Do method or maybe add some extra functionality to delete, this would make doing that refactor harder and really only save use 2 lines of very simple code

Reply:
the error handling I would like to leave until we do #3